src/core/misc.c

```diff
@@ -261,6 +261,26 @@ char *gslist_to_string(GSList *list, const char *delimiter)
 	return ret;
 }
 
+gboolean gslist_is_unique(GSList *list)
+{
+	const char *prev;
+
+	if (list == NULL || list->next == NULL)
+		return TRUE;
+
+	prev = (const char *)list->data;
+	list = list->next;
+
+	while (list != NULL) {
+		if (strcmp(list->data, prev) != 0)
+			return FALSE;
+		prev = (const char *)list->data;
+		list = list->next;
+	}
+
+	return FALSE;
```

Comment:
This is only meant to collapse the nicks in the trivial case when the same user does more than a single `MODE` change, could probably change the name of the function if you suggest a better one.